### PR TITLE
feat: add cache documentation

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What change does this PR introduce? 

adds usecase caching documantattion.

### Why was this change needed?

so the new and old users of the system will be aware of the new caching added and how to enable it.

### CODE REVIEW

File: docs/docs/overview/docker-deploy.md [hunk 1]
```diff
@@ -95,6 +95,18 @@ When using the IFrame embed to attach the notification center rather than the Re
 
 See [Use your own backend and socket URL](https://docs.novu.co/notification-center/react-components#use-your-own-backend-and-socket-url).
 
+### Caching
```
Comment (djabarovgeorge) [inline docs/docs/overview/docker-deploy.md:98]: I was not sure about the title here.

File: docs/docs/overview/docker-deploy.md [hunk 2]
```diff
@@ -95,6 +95,18 @@ When using the IFrame embed to attach the notification center rather than the Re
...
+
+We are introducing the first stage of caching in our system to improve performance and efficiency. Caching is turned off by default, but can easily be activated by setting the following environment variables:
```
Comment (djabarovgeorge) [inline docs/docs/overview/docker-deploy.md:100]: should we make the 'Caching is turned off by default' part more prominent somehow?

Comment (Cliftonz) [reply docs/docs/overview/docker-deploy.md:100]: I think we could specify that we are currently feature-flagging our caching implementation.
If a user has a use-case and wants to opt-in to cache they can turn it on by setting the following environment variables.

File: docs/docs/overview/docker-deploy.md (hunk 1 above)
Comment (Cliftonz) [reply docs/docs/overview/docker-deploy.md:98]: I think it is fine for the moment, this is usually just a baked-in feature and not really shown or talked about that often.

**Review by scopsy — APPROVED** [review]
I think what @Cliftonz makes sense. Other than that LGTM.

File: docs/docs/overview/docker-deploy.md (hunk 2 above)
Comment (djabarovgeorge) [reply docs/docs/overview/docker-deploy.md:100]: Hey @Cliftonz not sure if I got it there needed to be changes or not :), if there needed to be a change can you point me in the direction where you think we need, to add, or update some parts?

**Pull request merged**